Interface pull request, test unitaire et fin de l'interface des menus

MasterPNJ opened the pull request:
Tout les teste unitaires sur les menus ont été effectués et l'interface des menus sont fini.

MasterPNJ committed:
Ajout de theme

Ajout de theme : dark theme, candy theme

MasterPNJ committed:
Fix couleur du texte des boutons sur mac

Fix couleur du texte des boutons sur mac

MasterPNJ committed:
Fix des autres thèmes pour mac

Fix des autres thèmes pour mac

MasterPNJ committed:
Test unitaire pour le menu principal

Test unitaire pour le menu principal

MasterPNJ committed:
Fix d'import non-utile

Fix d'import non-utile

MasterPNJ committed:
Test unitaire des boutons du menu principal

Test unitaire des boutons du menu principal

Co-Authored-By: Antoine <antoine@enimaloc.fr>

MasterPNJ committed:
Ajout de la vérification de l'application du thème sur les boutons

Ajout de la vérification de l'application du thème sur les boutons

MasterPNJ committed:
Ajout test unitaire menu paramètre

Ajout test unitaire menu paramètre

MasterPNJ committed:
Optimisation test unitaire

Optimisation test unitaire

MasterPNJ committed:
Création des tests pour tout les autres menus ( WIP )

Création des tests pour tout les autres menus. ( WIP )

MasterPNJ committed:
Optimisation des tests des menus, suppression de la quasi totalité des warnings, ajout des commentaires

Optimisation des tests des menus, suppression de la quasi totalité des warnings, ajout des commentaires

HiraAkshayy committed:
Ajout du test du menu pour ProfilSelectionTest

HiraAkshayy committed:
Merge branch 'Interface' of https://github.com/projet-gl2/Hashimiste into Interface

HiraAkshayy committed:
Changements du test des themes

HiraAkshayy committed:
Légères modifications de commentaires

MasterPNJ committed:
Fix test du profilSelection au niveau du manque d'initialisation du menu + fix warning

Fix test du profilSelection au niveau du manque d'initialisation du menu qui empêcher la bon dérouler du test.
Fix de warning

enimaloc commented on src/test/java/fr/hashimiste/impl/gui/menu/AventureTest.java:
```diff
@@ -0,0 +1,68 @@
+import java.io.IOException;
+import java.util.*;
+
+class AventureTest extends fr.hashimiste.impl.gui.menu.Test{
```
utiliser la directive `import`

enimaloc commented on src/test/java/fr/hashimiste/impl/gui/menu/AventureTest.java:
```diff
@@ -0,0 +1,68 @@
+     * Teste le menu Paramètre
+     */
+    @Test
+    void TestMenuParametre(){
```
par convention les méthodes commence par une minuscule

enimaloc commented on src/test/java/fr/hashimiste/impl/gui/menu/MenuTest.java:
```diff
@@ -0,0 +1,114 @@
+import fr.hashimiste.core.joueur.Profil;
+import fr.hashimiste.core.gui.JFrameTemplate;
+
+class MenuTest extends fr.hashimiste.impl.gui.menu.Test {
```
utiliser la directive `import`

enimaloc commented on src/test/java/fr/hashimiste/impl/gui/menu/ModeLibreTest.java:
```diff
@@ -0,0 +1,59 @@
+import java.io.IOException;
+import java.util.*;
+
+class ModeLibreTest extends fr.hashimiste.impl.gui.menu.Test{
```
utiliser la directive `import`

enimaloc commented on src/test/java/fr/hashimiste/impl/gui/menu/ParametreTest.java:
```diff
@@ -0,0 +1,62 @@
+import java.io.IOException;
+import java.util.*;
+
+class ParametreTest extends fr.hashimiste.impl.gui.menu.Test{
```
utiliser la directive `import`

enimaloc commented on src/test/java/fr/hashimiste/impl/gui/menu/ProfilSelectionTest.java:
```diff
@@ -0,0 +1,78 @@
+import java.util.List;
+import java.util.Properties;
+
+class ProfilSelectionTest extends fr.hashimiste.impl.gui.menu.Test{
```
utiliser la directive `import`

enimaloc commented on src/test/java/fr/hashimiste/impl/gui/menu/TechniqueTest.java:
```diff
@@ -0,0 +1,57 @@
+import java.io.IOException;
+import java.util.*;
+
+class TechniqueTest extends fr.hashimiste.impl.gui.menu.Test{
```
utiliser la directive `import`

MasterPNJ committed:
Fix convention + import

Fix convention + import.
Changement du nom de fichier "Test" en "TestMenu" pour éviter les conflits.
